appserver/payara-appserver-modules/security-openid/src/main/java/fish/payara/security/openid/google/GoogleOpenIdExtension.java

```diff
@@ -202,6 +202,16 @@ public int jwksConnectTimeout() {
             public int jwksReadTimeout() {
                 return googleDefinition.jwksReadTimeout();
             }
+
+            @Override
+            public boolean tokenAutoRefresh() {
+                return false; // TODO check if "true" is working with Google
```

Comment:
You may fetch `refreshToken` with `access_type` extra param for testing : 
`@GoogleAuthenticationDefinition (
        clientId = "XXXXXXXXXXXXXXXapps.googleusercontent.com",
        clientSecret = "XXXXXXXXXXXX",
        extraParameters = {"access_type=offline"}
)`

Reply:
Unfortunately, I am currently not able to check/test this.